### PR TITLE
[HttpFoundation] Add Request::isMethodIdempotent method

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Branch? | master |
| Bug fix? | no |
| New feature? | yes |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

Addd a new method in the spirit of `isMethodSafe` to know if the current method is idempotent according to RFCs.

### CODE REVIEW

**Comment by dunglas** [comment]
AppVeyor failures unrelated.

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 1]
```diff
@@ -1483,6 +1483,24 @@ public function isMethodSafe()
...
+    public function isMethodIdempotent()
+    {
+        return in_array($this->getMethod(), array(
+            self::METHOD_HEAD,
+            self::METHOD_GET,
+            self::METHOD_PUT,
+            self::METHOD_DELETE,
+            self::METHOD_TRACE,
+            self::METHOD_OPTIONS,
+            self::METHOD_PURGE,
```
Comment (fabpot) [inline src/Symfony/Component/HttpFoundation/Request.php]: I would use the strings here directly as done for the `isMethodSafe` method.

File: src/Symfony/Component/HttpFoundation/Request.php [hunk 2]
```diff
@@ -1483,6 +1483,24 @@ public function isMethodSafe()
     }
 
     /**
+     * Checks whether the method is idempotent or not.
```
Comment (xabbuh) [inline src/Symfony/Component/HttpFoundation/Request.php]: I would go for "Checks whether or not the method is idempotent." or "Checks if the method is idempotent."

Comment (dunglas) [inline src/Symfony/Component/HttpFoundation/Request.php]: I used the same text than in isMethodSafe

**Comment by dunglas** [comment]
Comments fixed.

**Comment by fabpot** [comment]
Thank you @dunglas.

File: src/Symfony/Component/HttpFoundation/Request.php (hunk 1 above)
Comment (rodmarzavala) [inline src/Symfony/Component/HttpFoundation/Request.php]: Hi, why use strings instead of constants here?

**Pull request closed without merging**